File: 240711/3n + 1 수열 m번/3n-plus-1-sequence-m-times.js
const fs = require('fs');
const arr = fs.readFileSync(0).toString().trim().split('\n');

const n = Number(arr[0]);

for(let i = 1; i <= n; i++){
    console.log(getResult(Number(arr[i])));
}

function getResult(num){
    let cnt = 0;

    while(num !== 1){
        if(num % 2 === 0) num /= 2;
        else num = num * 3 + 1;
        cnt++;
    }

    return cnt;
}

File: 240711/3n + 1 수열 m번/3n-plus-1-sequence-m-times.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int m = sc.nextInt();

        for(int i = 0; i < m; i++){
            System.out.println(getResult(sc.nextInt()));
        }
    }

    public static int getResult(int num){
        int cnt = 0;

        while(num != 1){
            if(num % 2 == 0) num /= 2;
            else num = num * 3 + 1;
            cnt++;
        }

        return cnt;
    }
}